californium-core/src/main/java/org/eclipse/californium/core/network/RandomTokenGenerator.java

```diff
@@ -56,16 +58,52 @@ public RandomTokenGenerator(final NetworkConfig networkConfig) {
 	}
 
 	@Override
-	public Token createToken(boolean longTermScope) {
+	public Token createToken(Scope scope) {
 		byte[] token = new byte[tokenSize];
 		rng.nextBytes(token);
-		if (longTermScope) {
+		switch (scope) {
+		case LONG_TERM:
 			// set bit 0 to 1
-			token[0] |= 0x1;
-		} else {
-			// set bit 0 to 0
-			token[0] &= 0xfe;
+			token[0] |= 0b1;
+			break;
+		case SHORT_TERM:
+			// set bit 1-0 to 0b10
+			token[0] &= 0b11111100;
+			token[0] |= 0b10;
```

Comment:
<nitpicking>for mask, I feel it is a bit clearer to use the full notation like 0b00000010</nitpicking>

Reply:
That makes the compile time too long ;-)